src/chain/ethereum.rs

```diff
@@ -55,9 +62,28 @@ pub enum EthereumGasPriceError {
     MissingMaxFeePerGas,
 }
 
+fn get_ethereum_gas_price(tx: &Transaction) -> Result<U256, EthereumGasPriceError> {
+    let tx_type_raw: u8 = tx.transaction_type.unwrap_or_default();
+    let Ok(tx_type) = TxType::try_from(tx_type_raw) else {
+        return Err(EthereumGasPriceError::InvalidType(tx_type_raw));
+    };
+
+    match tx_type {
+        TxType::Legacy | TxType::Eip2930 => tx
+            .gas_price
+            .map(U256::from)
+            .ok_or(EthereumGasPriceError::MissingGasPrice),
+        TxType::Eip1559 | TxType::Eip4844 | TxType::Eip7702 => tx
+            .max_fee_per_gas
+            .map(U256::from)
+            .ok_or(EthereumGasPriceError::MissingMaxFeePerGas),
+    }
+}
+
 impl PevmChain for PevmEthereum {
+    type Transaction = Transaction;
```

Comment:
Rare place where we should add `::` to be honest.

Reply:
Can you check again? I explicit write `alloy_rpc_types::Transaction` in this place and some more places.